pkgs/tools/security/fprintd/default.nix

```diff
@@ -1,43 +1,67 @@
-{ thinkpad ? false
-, stdenv
-, fetchurl
+{ stdenv
+, fetchFromGitLab
 , fetchpatch
 , pkgconfig
-, intltool
-, libfprint-thinkpad ? null
-, libfprint ? null
+, meson
+, ninja
+, perl
+, gettext
+, cairo
+, gtk-doc
+, libxslt
+, docbook-xsl-nons
+, docbook_xml_dtd_412
 , glib
+, dbus
 , dbus-glib
 , polkit
 , nss
 , pam
 , systemd
-, autoreconfHook
-, gtk-doc
+, libfprint
+, python3
 }:
 
 stdenv.mkDerivation rec {
-  pname = "fprintd" + stdenv.lib.optionalString thinkpad "-thinkpad";
-  version = "0.9.0";
+  pname = "fprintd";
+  version = "1.90.1";
+  outputs = [ "out" "devdoc" ];
 
-  src = fetchurl {
-    url = "https://gitlab.freedesktop.org/libfprint/fprintd/uploads/9dec4b63d1f00e637070be1477ce63c0/fprintd-${version}.tar.xz";
-    sha256 = "182gcnwb6zjwmk0dn562rjmpbk7ac7dhipbfdhfic2sn1jzis49p";
+  src = fetchFromGitLab {
+    domain = "gitlab.freedesktop.org";
+    owner = "libfprint";
+    repo = pname;
+    rev = version;
+    sha256 = "0mbzk263x7f58i9cxhs44mrngs7zw5wkm62j5r6xlcidhmfn03cg";
   };
 
   patches = [
+    # Fixes issue with ":" used when multiple paths (might be the case on NixOS)
+    # https://gitlab.freedesktop.org/libfprint/fprintd/-/merge_requests/50
     (fetchpatch {
-      url = "https://gitlab.freedesktop.org/libfprint/fprintd/merge_requests/16.patch";
-      sha256 = "1y39zsmxjll9hip8464qwhq5qg06c13pnafyafgxdph75lvhdll7";
+      url = "https://gitlab.freedesktop.org/libfprint/fprintd/-/merge_requests/50.patch";
+      sha256 = "1wqkk7bqig0fhjbq1487kxkh7yc9356qjsac1x27pzgakbayxpyy";
     })
   ];
 
   nativeBuildInputs = [
-    intltool
     pkgconfig
-    autoreconfHook # Drop with above patch
-    gtk-doc # Drop with above patch
-  ];
+    meson
+    ninja
+    perl
+    gettext
+    gtk-doc
+    libxslt
+    dbus
+    docbook-xsl-nons
+    docbook_xml_dtd_412
+  ] ++ (with python3.pkgs; [
```

Comment:
Ideally, these would be in `checkInputs` but that would require `doCheck = true`. Maybe add a TODO comment?

Reply:
I added the TODO, I've tried to move it, but it breaks things. Out of curiosity, what would be the requirement to accomplish such modification ? I have the feeling that I should disable the tests from the buildPhase, and write a custom `checkPhase` right ?

Reply:
Ideally, you would figure out how to fix the tests – often it is just a matter of running them inside dbus session:

https://github.com/NixOS/nixpkgs/blob/4b4e2b390ae53e1213081504b201a8e0e2e566f7/pkgs/applications/networking/calls/default.nix#L69-L76

(not all these lines necessary)

but `checkPhase = ":";` (`:` = shell null command) with a `TODO` comment would work too.

Reply:
With the `checkPhase = ":"` it still fails with:

```
ERROR: Dependency "pam_wrapper" not found, tried pkgconfig
```

I will play with this next week-end if that's not a blocker :).